### PR TITLE
[xml] Fix Saxon not reading xml DOM attributes

### PR DESCRIPTION
**Pull request by oowekyala**
At some point I think it would be best to merge the AttributeNode interface into the Node interface.

### CODE REVIEW

File: pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/saxon/AttributeAxisIterator.java [hunk 1]
```diff
@@ -24,7 +26,9 @@
      */
     public AttributeAxisIterator(ElementNode startNodeInfo) {
         this.startNodeInfo = startNodeInfo;
-        this.iterator = new net.sourceforge.pmd.lang.ast.xpath.AttributeAxisIterator(startNodeInfo.node);
+        this.iterator = startNodeInfo.node instanceof net.sourceforge.pmd.lang.ast.xpath.AttributeNode
+                ? ((net.sourceforge.pmd.lang.ast.xpath.AttributeNode) startNodeInfo.node).getAttributeIterator()
+                : new net.sourceforge.pmd.lang.ast.xpath.AttributeAxisIterator(startNodeInfo.node);
```
Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/saxon/AttributeAxisIterator.java]: this exact logic is repeated the change you made to `NodeInfoPanelController` and already present in `DocumentNavigator.getAttributeAxisIterator`... it seems like we should refactor this away

Comment (oowekyala) [reply pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/saxon/AttributeAxisIterator.java]: Note that in the case of `getXPathNodeName()` we added it directly to the Node interface and provided a default implementation in AbstractNode, which was deemed compatible enough at the time. Doing the same thing here would make sense I think

Comment (jsotuyod) [reply pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/saxon/AttributeAxisIterator.java]: I support doing the same here, yes

Comment (oowekyala) [reply pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/saxon/AttributeAxisIterator.java]: Done!

File: pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AbstractNode.java [hunk 2]
```diff
@@ -515,4 +516,10 @@ public String getXPathNodeName() {
     public String toString() {
         return getXPathNodeName();
     }
+
+
+    @Override
+    public Iterator<Attribute> getXPathAttributes() {
```
Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AbstractNode.java]: `getXPathAttributeIterator()` would be more descriptive, wouldn't?

File: pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/Node.java [hunk 3]
```diff
@@ -322,4 +324,12 @@
      * @return The XPath node name
      */
     String getXPathNodeName();
+
+
+    /**
+     * Returns an iterator containing all the attributes that are available
+     * from XPath for this node.
+     */
+    Iterator<Attribute> getXPathAttributes();
```
Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/Node.java]: the javadoc is missing the `@return` tag and will produce a warning

**Pull request merged**